### ISSUE TITLE
Closing the trade window causes exception when getting user's display name

### ISSUE THREAD

**Issue by thomotron** [thomotron/Phinix#5]
### Description
The trade backend seems to delete the trade before the window can close, causing it to try get the display name from a UUID that has been null'd

```
Exception filling window for PhinixClient.TradeWindow: System.ArgumentException: UUID cannot be null or empty.  
Parameter name: uuid  
  at UserManagement.UserManager.TryGetDisplayName (System.String uuid, System.String& displayName) [0x00000] in <filename unknown>:0  
  at PhinixClient.Client.TryGetDisplayName (System.String uuid, System.String& displayName) [0x00000] in <filename unknown>:0  
  at PhinixClient.TradeWindow.GetOtherPartyDisplayName () [0x00000] in <filename unknown>:0 
  at PhinixClient.TradeWindow.GenerateOffers () [0x00000] in <filename unknown>:0  
  at PhinixClient.TradeWindow.DoWindowContents (Rect inRect) [0x00000] in <filename unknown>:0 
  at Verse.Window+<WindowOnGUI>c__AnonStorey0.<>m__0 (Int32 x) [0x00000] in <filename unknown>:0
```

### Steps to reproduce
1. Start a trade
2. Open the trade window
3. Cancel it from the other client
4. The console should pop up with something like the above

**Comment by thomotron** [comment]
This seems to be because there's more than one thread that updates the trade window. The update would come in, mark the trade as complete, and close the window. All while the main GUI thread is still trying to render the item list for example.

**Comment by thomotron** [comment]
This has effectively been fixed by baacc2a2 as the trade window will render from a snapshot of the trade's state.

If a similar issue occurs, then a new issue should be opened to cover it.
